pepper-apis/src/main/java/org/broadinstitute/ddp/route/PatchFormAnswersRoute.java

```diff
@@ -168,7 +170,7 @@ public Object handle(Request request, Response response) {
                 return res;
             }
 
-            if (ActivityInstanceUtil.isReadonly(formActivityDef.getEditTimeoutSec(), instanceDto.getCreatedAtMillis(),
+            if (!isStudyAdmin && ActivityInstanceUtil.isReadonly(formActivityDef.getEditTimeoutSec(), instanceDto.getCreatedAtMillis(),
```

Comment:
I wonder if we should add an `isStudyAdmin` param to various `isReadonly()` methods?  This would centralize the decision about whether a thing is read-only given the additional context of the user.

Reply:
From what I can tell, `ActivityInstanceUtil.isReadonly()` is used in lower-level DAO methods. I'm hesitate to pass in `isStudyAdmin` all the way down there because:

1. It will need to be pass through several layers, resulting in a handful of refactors.
2. And most importantly, I don't think these two should be tied together. "read-only" is a property of the activity instance, and so the computation for that property ideally shouldn't involved other things like the authentication status. `isStudyAdmin` is really just a way for us to bypass that "read-only" property -- the activity instance itself is still inherently read-only.